### DISCUSSION TITLE
Restoring SeedVault backup from Pixel 6 Pro on a new Pixel 8 Pro

### DISCUSSION THREAD

**Discussion by deanbuxton** [seedvault-app/seedvault#587] [General]
I am trying to backup a Pixel 6a Pro phone using SeedVault to USB-C flash drive and restore that to a new Pixel 8 Pro. However, when I attempt to restore on the new phone, it cannot see or detect backup file from the old phone on the flash drive. I have to my missing something easy. I tried to restore during first boot on the new phone, can't see backup. Next I entered recovery mode on the new phone, still can't see backup.

On the Pixel 8 Pro, I rebooted to 'Recovery Mode' which when entered eventually boots you 'Safe Mode' it says on the bottom of the screen. When I enter the 'random 12 keywords' from either the old or new phone, does not see the back file from the Pixel 6 Pro phone.

Each phone can independently perform its own backup to the USB-C flash drive. They just can't seem to find the magic process to get it to see the other phone's backup file. Someone put me out of my misery.

++++++++++++++++++++++++++++++++++++
Restoring a backup during first time setup/boot on new GrapheneOS Pixel 8 Pro phone:
++++++++++++++++++++++++++++++++++++

Backup on Old Pixel Phone: to USB-C flash drive

Install GrapheneOS SeedVault:

Ensure that GrapheneOS SeedVault is installed and enabled on your old Pixel phone. You can find it in the Apps section of your device settings.
Configure SeedVault:

Open SeedVault from your app list.
Set up the backup destination. You can use an external USB drive, a network share, or a cloud service (if you trust it with your data).
Initiate Backup:

In SeedVault, initiate a backup of your device. This will include your apps, app data, and device settings.
Encryption Key:

SeedVault will create an encryption key for your backup. It's essential to remember this key or store it securely because you'll need it for the restore process.
Backup Process:

SeedVault will begin the backup process. It might take some time, depending on the amount of data.
Verify Backup:

Once the backup is complete, verify that your data is securely stored in the chosen destination.

Restore on New Pixel Phone:

Install GrapheneOS SeedVault:

Make sure that you have GrapheneOS SeedVault installed on your new Pixel phone. You can find it in the Apps section of your device settings.
Initialize Device:

Set up your new Pixel phone, but stop at the Wi-Fi network setup screen.
Connect to Backup:

On the Wi-Fi setup screen, connect your new phone to the same network as your old phone.
Restore from Backup:

You will be prompted to restore from a backup. Select this option.
Select Source:

Choose the source of your backup, which could be a USB drive, network share, or cloud service where you saved your old phone's backup.
(This is where mine cannot see the old backup from the other phone)
Enter Encryption Key:

Enter the encryption key you created during the backup process on your old phone. This is crucial to decrypt your data.
Restore Process:

SeedVault will start restoring your data, including apps, app data, and settings, to your new Pixel phone.
Complete Setup:

Once the restore is finished, complete the initial setup of your new phone, including configuring your Google account or any other necessary settings.
Verification:

Verify that your data has been successfully transferred to your new Pixel phone.

+++++++++++++++++++++++++++++++++
Performing a restore with GrapheneOS SeedVault on a new Pixel phone that has already been initialized can be a bit more complex, but it is still possible.
Here's a step-by-step guide on how to do it:
+++++++++++++++++++++++++++++++++
Restoring Backup on a New Pixel Phone After Initialization:

Install GrapheneOS SeedVault:

Ensure that GrapheneOS SeedVault is installed and enabled on your new Pixel phone. You can find it in the Apps section of your device settings.
Connect to the Backup Location:

You'll need access to the location where you stored your old Pixel phone's backup. This can be an external USB drive, a network share, or a cloud service (if you trust it with your data).
Enter Recovery Mode:

To restore your backup, you might need to enter the device's recovery mode. The steps to enter recovery mode can vary by device, so check your device's documentation or online resources for specific instructions.
Wipe Data:

In recovery mode, you should perform a data wipe or factory reset on your new Pixel phone. This step is essential to ensure that the device is in a state where it can accept the backup. The exact process for doing this can differ depending on your device.
Connect to Backup Source:

In recovery mode, there is usually an option to connect to an external storage source. Use this option to access the backup location where your old Pixel phone's data is stored.
Restore Backup:

Select the option to restore a backup from the connected source. You might be prompted to navigate to the location of your backup and select it.
Enter Encryption Key:

You will be asked to enter the encryption key that you created during the backup process on your old Pixel phone. This key is necessary to decrypt your data.
Restore Process:

SeedVault will begin the process of restoring your data, including apps, app data, and settings, to your new Pixel phone.
Complete Setup:

Once the restore is finished, complete the initial setup of your new phone, including configuring your Google account or any other necessary settings.
Verification:

Verify that your data has been successfully transferred and restored to your new Pixel phone.

**Comment by grote** [comment]
On the flashdrive is there a `.Seedvault*` folder? how big is it? 

For restore, it needs to use the same recovery code. Could it be that on both phones, Seedvault was already set up with different recovery codes? Restore is currently only support during device provisioning via initial SetupWizard.
